central/authprovider/service/service_impl.go

```diff
@@ -195,7 +209,9 @@ func (s *serviceImpl) PutAuthProvider(ctx context.Context, request *storage.Auth
 		return nil, err
 	}
 
-	provider, err := s.registry.CreateProvider(ctx, authproviders.WithStorageView(request), authproviders.WithValidateCallback(datastore.Singleton()))
+	provider, err := s.registry.CreateProvider(ctx, authproviders.WithStorageView(request),
+		authproviders.WithAttributeChecker(request),
```

Comment:
Seems like `authproviders.WithAttributeChecker(request)` would not do anything

Reply:
Not entirely true for the Put case.

Imagine we have the RH SSO auth provider within managed services, and customer's decide to change it within the UI, this would trigger this re-creation. We have the required attributes sent from the FE to BE within the API request, and would need to re-create the auth provider with the attribute checker, since in this case the required attributes would be set -- otherwise, you're right, this would not be required.
If we were to delete this now, upon changing the initial provider, customers will get a different behavior after a Put operation.

Reply:
Just wondering, could we check if the current user would lock themselves out of the system? This is in case a new required attributed is added that is not actually provided by the identity provider. Perhaps that would even be the desired behaviour, but seems like a potentially bad UX.

Reply:
For making changes via `PutProvider`, the provider that will be updated has to be non-active, so the user wouldn't be _immediately_ locked out but would have the possibility to do a i.e. a test login after adding new settings.